src/Cardano/Shell/Configuration/Lib.hs

```diff
@@ -89,16 +89,31 @@ finaliseCardanoConfiguration PartialCardanoConfiguration{..} = do
     finaliseCore :: PartialCore -> Either Text Core
     finaliseCore PartialCore{..} = do
 
-        coGenesisFile               <- lastToEither "Unspecified coGenesisFile"             pcoGenesisFile
-        coGenesisHash               <- lastToEither "Unspecified coGenesisHash"             pcoGenesisHash
+        coGenesisFile                   <- lastToEither "Unspecified coGenesisFile"
+                                            pcoGenesisFile
 
-        coStaticKeySigningKeyFile   <- lastToEither "Unspecified coStaticKeySigningKeyFile" pcoStaticKeySigningKeyFile
-        coStaticKeyDlgCertFile      <- lastToEither "Unspecified coStaticKeyDlgCertFile"    pcoStaticKeyDlgCertFile
+        coGenesisHash                   <- lastToEither "Unspecified coGenesisHash"
+                                            pcoGenesisHash
 
-        coRequiresNetworkMagic      <- lastToEither "Unspecified coRequiresNetworkMagic"    pcoRequiresNetworkMagic
-        coDBSerializeVersion        <- lastToEither "Unspecified coDBSerializeVersion"      pcoDBSerializeVersion
+        coNodeId                        <- lastToEither "Unspecified coNodeId"
+                                            pcoNodeId
 
-        coPBftSigThd                <- lastToEither "Unspecified coPBftSigThd"              pcoPBftSigThd
+        coNumCoreNodes                  <- lastToEither "Unspecified coNumCoreNodes"
+                                            pcoNumCoreNodes
```

Comment:
Ah, this can't be a required param. It turns out that this param is only used for some protocols, and not the production PBFT one. It's used only for the simple demo protocols.

I think the right thing to do is to make `coNumCoreNodes` itself by a `Maybe` and treat it like we do for `pcoStaticKeySigningKeyFile` using `getLast`.